Failure to build on 7.1-rc1

```
sudo ./install-driver.sh
Please copy and post all displayed lines when reporting an issue!
Press Enter to continue...

  ================================================================
     mt76 WiFi Driver Installer
  ================================================================
  install-driver.sh v20260408
  Source:  90ab9ae0
  Distro:  Ubuntu 26.04 LTS
  Kernel:  7.1.0-rc1 (x86_64)
  Params:   ro, mitigations=off, i915.mitigations=off, acpi_mask_gpe=0x17, rcuupdate.rcu_cpu_stall_suppress=1, zswap.enabled=1, zswap.zpool=zsmalloc, log_buf_len=1M, nouveau.config=NvPmEnableGating=3, zfs.zil.replay_disable=1, zfs.zfs_recover=1, cfi=off, nouveau.atomic=1
  CPUs:    8/8 (in-use/total)
  Memory:  16279380 kB
  gcc:     gcc (Ubuntu 15.2.0-16ubuntu1) 15.2.0
  dkms:    dkms-3.2.2
  SecBoot: not supported
  Reg:     country US: DFS-FCC

  Detected mt76 hardware:
    mt7925u: ID 0846:9072 MediaTek Inc. Wireless_Device
  ----------------------------------------------------------------

[1/7] Checking prerequisites
  All prerequisites found.
  Kernel headers present.

[2/7] Removing previous installation (if any)
  Clean.

[3/7] Configuring
  Installing mt76_git.conf to /etc/modprobe.d
  Done.

[4/7] Installing firmware
  Installed 23 firmware file(s) to /lib/firmware/mediatek

[5/7] Building modules
  (dkms build)
  Copying source to /usr/src/mt76-1.0
Creating symlink /var/lib/dkms/mt76/1.0/source -> /usr/src/mt76-1.0
  Added to dkms.
Sign command: /usr/bin/kmodsign
Signing key: /var/lib/shim-signed/mok/MOK.priv
Public certificate (MOK): /var/lib/shim-signed/mok/MOK.der

Building module(s).........(bad exit status: 2)
Failed command:
KVER=7.1.0-rc1 'make' modules
ERROR (dkms apport): binary package for mt76: 1.0 not found

Error! Bad return status for module build on kernel: 7.1.0-rc1 (x86_64)
Consult /var/lib/dkms/mt76/1.0/build/make.log for more information.
Command exited with non-zero status 10
  Compile time: 108.66 seconds
An error occurred: 10
Please report this error.
Please copy and post the following items into the problem report.
    -all screen output from install-driver.sh
You should run the following before reattempting installation.
$ sudo ./uninstall-driver.sh
```

`/var/lib/dkms/mt76/1.0/build/make.log`:
```
DKMS (dkms-3.2.2) make.log for mt76/1.0 for kernel 7.1.0-rc1 (x86_64)
Tue Apr 28 01:56:21 PM EDT 2026

Building module(s)
# command: KVER=7.1.0-rc1 'make' modules
make -j7 -C /lib/modules/7.1.0-rc1/build M=$PWD modules
make[1]: Entering directory '/usr/src/linux-headers-7.1.0-rc1'
make[2]: Entering directory '/var/lib/dkms/mt76/1.0/build'
warning: pahole version differs from the one used to build the kernel
  The kernel was built with: 131
  You are using:             0
  CC [M]  mmio.o
  CC [M]  util.o
  CC [M]  trace.o
  CC [M]  dma.o
  CC [M]  mac80211.o
  CC [M]  debugfs.o
  CC [M]  eeprom.o
  CC [M]  tx.o
  CC [M]  agg-rx.o
  CC [M]  mcu.o
  CC [M]  wed.o
  CC [M]  scan.o
  CC [M]  channel.o
  CC [M]  pci.o
  CC [M]  usb.o
  CC [M]  usb_trace.o
  CC [M]  sdio.o
  CC [M]  sdio_txrx.o
  CC [M]  mt76x02_util.o
  CC [M]  mt76x02_mac.o
  CC [M]  mt76x02_mcu.o
  CC [M]  mt76x02_eeprom.o
  CC [M]  mt76x02_phy.o
  CC [M]  mt76x02_mmio.o
  CC [M]  mt76x02_txrx.o
  CC [M]  mt76x02_trace.o
  CC [M]  mt76x02_debugfs.o
  CC [M]  mt76x02_dfs.o
  CC [M]  mt76x02_beacon.o
  CC [M]  mt76x02_usb_mcu.o
  CC [M]  mt76x02_usb_core.o
  CC [M]  mt76_connac_mcu.o
  CC [M]  mt76_connac_mac.o
  CC [M]  mt76_connac3_mac.o
  CC [M]  mt792x_core.o
mt76_connac_mac.c: In function ‘mt76_connac2_mac_write_txwi_80211’:
mt76_connac_mac.c:416:25: error: ‘IEEE80211_MIN_ACTION_SIZE’ undeclared (first use in this function)
  416 |             skb->len >= IEEE80211_MIN_ACTION_SIZE + 1 &&
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~
mt76_connac_mac.c:416:25: note: each undeclared identifier is reported only once for each function it appears in
mt76_connac_mac.c:418:27: error: ‘struct <anonymous>’ has no member named ‘u’
  418 |             mgmt->u.action.u.addba_req.action_code == WLAN_ACTION_ADDBA_REQ) {
      |                           ^
In file included from /usr/src/linux-headers-7.1.0-rc1/include/linux/byteorder/little_endian.h:5,
                 from /usr/src/linux-headers-7.1.0-rc1/arch/x86/include/uapi/asm/byteorder.h:5,
                 from /usr/src/linux-headers-7.1.0-rc1/include/asm-generic/qrwlock_types.h:6,
                 from /usr/src/linux-headers-7.1.0-rc1/arch/x86/include/asm/spinlock_types.h:7,
                 from /usr/src/linux-headers-7.1.0-rc1/include/linux/spinlock_types_raw.h:7,
                 from /usr/src/linux-headers-7.1.0-rc1/include/linux/ratelimit_types.h:7,
                 from /usr/src/linux-headers-7.1.0-rc1/include/linux/printk.h:9,
                 from /usr/src/linux-headers-7.1.0-rc1/include/asm-generic/bug.h:31,
                 from /usr/src/linux-headers-7.1.0-rc1/arch/x86/include/asm/bug.h:193,
                 from /usr/src/linux-headers-7.1.0-rc1/arch/x86/include/asm/alternative.h:9,
                 from /usr/src/linux-headers-7.1.0-rc1/arch/x86/include/asm/barrier.h:5,
                 from /usr/src/linux-headers-7.1.0-rc1/include/asm-generic/bitops/generic-non-atomic.h:7,
                 from /usr/src/linux-headers-7.1.0-rc1/include/linux/bitops.h:28,
                 from /usr/src/linux-headers-7.1.0-rc1/include/linux/kernel.h:23,
                 from mt76.h:9,
                 from mt76_connac.h:7,
                 from mt76_connac_mac.c:4:
mt76_connac_mac.c:419:55: error: ‘struct <anonymous>’ has no member named ‘u’
  419 |                 u16 capab = le16_to_cpu(mgmt->u.action.u.addba_req.capab);
      |                                                       ^
/usr/src/linux-headers-7.1.0-rc1/include/uapi/linux/byteorder/little_endian.h:37:51: note: in definition of macro ‘__le16_to_cpu’
   37 | #define __le16_to_cpu(x) ((__force __u16)(__le16)(x))
      |                                                   ^
mt76_connac_mac.c:419:29: note: in expansion of macro ‘le16_to_cpu’
  419 |                 u16 capab = le16_to_cpu(mgmt->u.action.u.addba_req.capab);
      |                             ^~~~~~~~~~~
  CC [M]  mt792x_mac.o
make[4]: *** [/usr/src/linux-headers-7.1.0-rc1/scripts/Makefile.build:289: mt76_connac_mac.o] Error 1
make[4]: *** Waiting for unfinished jobs....
make[3]: *** [/usr/src/linux-headers-7.1.0-rc1/Makefile:2141: .] Error 2
make[2]: *** [/usr/src/linux-headers-7.1.0-rc1/Makefile:248: __sub-make] Error 2
make[2]: Leaving directory '/var/lib/dkms/mt76/1.0/build'
make[1]: *** [Makefile:248: __sub-make] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-7.1.0-rc1'
make: *** [Makefile:114: modules] Error 2

# exit code: 2
# elapsed time: 00:00:18
----------------------------------------------------------------
```

--- a/mt76_connac_mac.c
+++ b/mt76_connac_mac.c
@@ -412,11 +412,23 @@ mt76_connac2_mac_write_txwi_80211(struct mt76_dev *dev, __le32 *txwi,
 	u8 fc_type, fc_stype;
 	u32 val;
 
+/* compat: in kernel 7.1 IEEE80211_MIN_ACTION_SIZE became a function-like
+ * macro, the inner u.action.u union was flattened, and action_code moved
+ * to the action level instead of being a per-action-struct field.
+ */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0)
+	if (ieee80211_is_action(fc) &&
+	    skb->len >= IEEE80211_MIN_ACTION_SIZE(addba_req) &&
+	    mgmt->u.action.category == WLAN_CATEGORY_BACK &&
+	    mgmt->u.action.action_code == WLAN_ACTION_ADDBA_REQ) {
+		u16 capab = le16_to_cpu(mgmt->u.action.addba_req.capab);
+#else
 	if (ieee80211_is_action(fc) &&
 	    skb->len >= IEEE80211_MIN_ACTION_SIZE + 1 &&
 	    mgmt->u.action.category == WLAN_CATEGORY_BACK &&
 	    mgmt->u.action.u.addba_req.action_code == WLAN_ACTION_ADDBA_REQ) {
 		u16 capab = le16_to_cpu(mgmt->u.action.u.addba_req.capab);
+#endif
 
 		txwi[5] |= cpu_to_le32(MT_TXD5_ADD_BA);
 		tid = (capab >> 2) & IEEE80211_QOS_CTL_TID_MASK;

--- a/mt7925/mac.c
+++ b/mt7925/mac.c
@@ -667,10 +667,18 @@ mt7925_mac_write_txwi_80211(struct mt76_dev *dev, __le32 *txwi,
 	u8 fc_type, fc_stype;
 	u32 val;
 
+/* compat: see note in mt76_connac_mac.c -- 7.1 mac80211 action API change */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0)
+	if (ieee80211_is_action(fc) &&
+	    skb->len >= IEEE80211_MIN_ACTION_SIZE(addba_req) &&
+	    mgmt->u.action.category == WLAN_CATEGORY_BACK &&
+	    mgmt->u.action.action_code == WLAN_ACTION_ADDBA_REQ)
+#else
 	if (ieee80211_is_action(fc) &&
 	    skb->len >= IEEE80211_MIN_ACTION_SIZE + 1 &&
 	    mgmt->u.action.category == WLAN_CATEGORY_BACK &&
 	    mgmt->u.action.u.addba_req.action_code == WLAN_ACTION_ADDBA_REQ)
+#endif
 		tid = MT_TX_ADDBA;
 	else if (ieee80211_is_mgmt(hdr->frame_control))
 		tid = MT_TX_NORMAL;

--- a/mt7996/mac.c
+++ b/mt7996/mac.c
@@ -762,10 +762,18 @@ mt7996_mac_write_txwi_80211(struct mt7996_dev *dev, __le32 *txwi,
 	u8 fc_type, fc_stype;
 	u32 val;
 
+/* compat: see note in mt76_connac_mac.c -- 7.1 mac80211 action API change */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0)
+	if (ieee80211_is_action(fc) &&
+	    skb->len >= IEEE80211_MIN_ACTION_SIZE(addba_req) &&
+	    mgmt->u.action.category == WLAN_CATEGORY_BACK &&
+	    mgmt->u.action.action_code == WLAN_ACTION_ADDBA_REQ) {
+#else
 	if (ieee80211_is_action(fc) &&
 	    skb->len >= IEEE80211_MIN_ACTION_SIZE + 1 &&
 	    mgmt->u.action.category == WLAN_CATEGORY_BACK &&
 	    mgmt->u.action.u.addba_req.action_code == WLAN_ACTION_ADDBA_REQ) {
+#endif
 		if (is_mt7990(&dev->mt76))
 			txwi[6] |= cpu_to_le32(FIELD_PREP(MT_TXD6_TID_ADDBA, tid));
 		else

